File: Models/SheetType.swift
import Foundation

// MARK: - Sheet Type Enum
enum SheetType: Identifiable, Equatable {
    case questions(QuestionCategory)
    case menu
    case subscription
    case favorites
    case journal
    case widgets
    case widgetTutorial
    case partnerManagement
    case locationPermission
    case partnerLocationMessage
    case eventsMap
    
    var id: String {
        switch self {
        case .questions(let category):
            return "questions_\(category.id)"
        case .menu:
            return "menu"
        case .subscription:
            return "subscription"
        case .favorites:
            return "favorites"
        case .journal:
            return "journal"
        case .widgets:
            return "widgets"
        case .widgetTutorial:
            return "widgetTutorial"
        case .partnerManagement:
            return "partnerManagement"
        case .locationPermission:
            return "locationPermission"
        case .partnerLocationMessage:
            return "partnerLocationMessage"
        case .eventsMap:
            return "eventsMap"
        }
    }
    
    static func == (lhs: SheetType, rhs: SheetType) -> Bool {
        switch (lhs, rhs) {
        case (.questions(let lhsCategory), .questions(let rhsCategory)):
            return lhsCategory.id == rhsCategory.id
        case (.menu, .menu),
             (.subscription, .subscription),
             (.favorites, .favorites),
             (.journal, .journal),
             (.widgets, .widgets),
             (.widgetTutorial, .widgetTutorial),
             (.partnerManagement, .partnerManagement),
             (.locationPermission, .locationPermission),
             (.partnerLocationMessage, .partnerLocationMessage),
             (.eventsMap, .eventsMap):
            return true
        default:
            return false
        }
    }
} 

File: Models/SheetType.kt
// SheetType.kt
package com.love2loveapp.ui.model

// Représente une catégorie (utilise ta classe existante si tu en as déjà une)
data class QuestionCategory(
    val id: String
    // ... autres champs si besoin (title, emoji, etc.)
)

/**
 * Équivalent Kotlin de l'enum Swift `SheetType: Identifiable, Equatable`.
 * - id stable pour chaque feuille (utile pour clés, navigation, etc.).
 * - Égalité :
 *   - Questions : compare uniquement category.id (comme le switch Equatable Swift).
 *   - Les autres sont des singletons object → égalité référentielle (mêmes cases).
 */
sealed interface SheetType {
    val id: String

    class Questions(val category: QuestionCategory) : SheetType {
        override val id: String = "questions_${category.id}"

        // Égalité basée uniquement sur l'id de la catégorie (mimique du Swift)
        override fun equals(other: Any?): Boolean =
            other is Questions && other.category.id == this.category.id

        override fun hashCode(): Int = category.id.hashCode()
        override fun toString(): String = "SheetType.Questions(id=$id)"
    }

    object Menu : SheetType { override val id: String = "menu" }
    object Subscription : SheetType { override val id: String = "subscription" }
    object Favorites : SheetType { override val id: String = "favorites" }
    object Journal : SheetType { override val id: String = "journal" }
    object Widgets : SheetType { override val id: String = "widgets" }
    object WidgetTutorial : SheetType { override val id: String = "widgetTutorial" }
    object PartnerManagement : SheetType { override val id: String = "partnerManagement" }
    object LocationPermission : SheetType { override val id: String = "locationPermission" }
    object PartnerLocationMessage : SheetType { override val id: String = "partnerLocationMessage" }
    object EventsMap : SheetType { override val id: String = "eventsMap" }
    object LocationTutorial : SheetType { override val id: String = "locationTutorial" }
    object DailyQuestionPermission : SheetType { override val id: String = "dailyQuestionPermission" }
}
